### ISSUE TITLE
Allow middleware on missing accept header

### ISSUE THREAD

**Issue by Xerkus** [weierophinney/problem-details#5]
Missing Accept header implies `Accept: */*`

I see two issues how package treats it:
- Response factory currently throws on empty Accept header value (in content negotiation dependency)
  https://github.com/weierophinney/problem-details/blob/master/src/ProblemDetailsResponseFactory.php#L289
  ```
  $mediaType = (new Negotiator())->getBest($accept, self::NEGOTIATION_PRIORITIES);
  ```
- And middleware acts as a noop. It looks like intentional move, but i fail to see the reasoning, browsers send Accept header by default and most of the clients that do not are api consumers.
  https://github.com/weierophinney/problem-details/blob/master/src/ProblemDetailsMiddleware.php#L59

**Comment by weierophinney** [comment]
> Response factory currently throws on empty Accept header value

Throws what?

Also, that helped me realize I was passing a second, default argument to `getHeaderLine()`... which doesn't do anything with it. The line pulling the `Accept` header should be more properly using the short ternary:

```php
$accept = $request->getHeaderLine('Accept') ?: 'application/xhtml+xml';
```

I'll get that fixed, which may fix the issue you were indicating.

> middleware acts as a noop. It looks like intentional move, but i fail to see the reasoning, browsers send Accept header by default and most of the clients that do not are api consumers.

If it is unable to provide an acceptable representation, then it will act as a no-op. The recommendation is not to use this as a general-purpose error handler, but one for APIs specifically. If the API is accessed in such a way that we cannot provide an acceptable representation, we should not.

That said, I'm amenable to modifying the middleware's `canActAsErrorHandler()` method to use `*/*` if the `Accept` header is empty — which, with the other change I propose above, would make it internally consistent.

**Comment by weierophinney** [comment]
Fixed with 9aaa282, and release 0.2.1.

**Comment by Xerkus** [comment]
I should have explained things better. You got it exactly right.

btw, this condition is unreachable now https://github.com/weierophinney/problem-details/blob/master/src/ProblemDetailsMiddleware.php#L58-L60
```
if (empty($accept)) {
    return false;
}
```
